cpp/arcticdb/storage/test/test_storage_exceptions.cpp

```diff
@@ -266,9 +252,63 @@ TEST(S3MockStorageTest, TestUnexpectedS3ErrorException ) {
     auto storage = factory.create();
 
     std::string failureSymbol = s3::MockS3Client::get_failure_trigger("sym1", s3::S3Operation::GET, Aws::S3::S3Errors::NETWORK_CONNECTION, false);
-    AtomKeyImpl k = entity::atom_key_builder().gen_id(0).build<entity::KeyType::VERSION>(failureSymbol);
 
     ASSERT_THROW({
-        storage->read(k, storage::ReadKeyOpts{});
+        read_in_store(*storage, failureSymbol);
     },  UnexpectedS3ErrorException);
 }
+
+// Azure error testing with mock client
+TEST(AzureMockStorageTest, TestReadKeyNotFoundException) {
+    AzureMockStorageFactory factory;
+    auto storage = factory.create();
+
+    ASSERT_TRUE(!exists_in_store(*storage, "sym"));
```

Comment:
Double negative! Should be `ASSERT_FALSE(exists_in_store...)`

Reply:
Fixed in [4ea3442](https://github.com/man-group/ArcticDB/pull/1344/commits/4ea3442882bfdc4001ff117531804a8ba28a8d08)